tests/ScopeTest.php

```diff
@@ -177,6 +178,42 @@ public function testContitionToWords()
         $this->assertEquals('Country that has reference Users where any referenced record exists', $country->scope()->toWords());
     }
 
+    public function testContitionUnsupportedToWords()
+    {
+        $condition = new Condition('name', 'abc');
+
+        $this->expectException(Exception::class);
+
+        $condition->toWords();
+    }
+
+    public function testContitionUnsupportedOperator()
+    {
+        $country = clone $this->country;
+
+        $this->expectException(Exception::class);
+
+        $country->addCondition('name', '==', 'abc');
+    }
+
+    public function testContitionUnsupportedNegate()
+    {
+        $condition = new Condition(new Expression('false'));
+
+        $this->expectException(Exception::class);
+
+        $condition->negate();
```

Comment:
Hmm - is there currently a way to negate expr?

Reply:
At the moment not so it throws exception. We can work on a property `$negate` to be set to `true` if there is no operator and then prefix the query with `NOT`

Reply:
I think is is quite major design limitation, final version of negate should be very carefull how we implement null handling (like does not in() match etc.)

what is not behaviour of `NULL IN(NULL)`?

Reply:
It is not a design limitation but a matter of implementation. Design allows it:)
Presently you have to add the `NOT` in your Expression (which basically raw sql).

Reply:
yes, current implementation limitation